test/powershell/Modules/Microsoft.PowerShell.Utility/Write-Host.Tests.ps1

```diff
@@ -1,18 +1,91 @@
-﻿Describe "Write-Host DRT Unit Tests" -Tags "CI" {
+﻿Describe "Write-Host DRT Unit Tests" -Tags "Slow","Feature" {
 
     BeforeAll {
         $powershell = Join-Path -Path $PsHome -ChildPath "powershell"
+
+        $testData = @(
+            @{ Name = '-Separator';       Command = "Write-Host a,b,c -Separator '+'";                 returnCount = 1; returnValue = @("a+b+c") }
+            @{ Name = '-NoNewline=true';  Command = "Write-Host a,b -NoNewline:`$true;Write-Host a,b"; returnCount = 1; returnValue = @("a ba b") }
+            @{ Name = '-NoNewline=false'; Command = "Write-Host a1,b1;Write-Host a2,b2";               returnCount = 2; returnValue = @("a1 b1","a2 b2") }
+        )
+    }
+
+    It "write-Host works with '<Name>' switch" -TestCases:$testData -Pending:$IsOSX {
+        param($Command, $returnCount, $returnValue)
+
+        [array]$result = & $powershell -noprofile $Command
+        $result.Count | Should Be $returnCount
+        $result[0] | Should Be $returnValue[0]
+        $result[1] | Should Be $returnValue[1]
```

Comment:
It's unlikely new test data will be aded to `$testData`, but I think it's better to use the same `foreach ($i in 0..($returnCount - 1))` pattern (or something similar).